qa/integration-tests/src/test/java/io/camunda/it/client/FlowNodeInstanceQueryTest.java

```diff
@@ -0,0 +1,492 @@
+        all.stream().sorted(Comparator.nullsLast(Comparator.naturalOrder())).toList();
+    final var sortedDesc =
+        all.stream().sorted(Comparator.nullsLast(Comparator.reverseOrder())).toList();
+
+    // then
+    assertThat(resultAsc.items().stream().map(FlowNodeInstance::getType).toList())
+        .containsExactlyElementsOf(sortedAsc);
+    assertThat(resultDesc.items().stream().map(FlowNodeInstance::getType).toList())
+        .containsExactlyElementsOf(sortedDesc);
+  }
+
+  @Test
+  void shouldQueryByTreePath() {
+    // given
+    final var treePath = flowNodeInstance.getTreePath();
+    // when
+    final var result =
+        zeebeClient.newFlownodeInstanceQuery().filter(f -> f.treePath(treePath)).send().join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(result.items().getFirst().getTreePath()).isEqualTo(treePath);
+  }
+
+  @Test
+  void shouldQueryByTenantId() {
+    // given
+    final var tenantId = flowNodeInstance.getTenantId();
+    // when
+    final var result =
+        zeebeClient.newFlownodeInstanceQuery().filter(f -> f.tenantId(tenantId)).send().join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(7);
+    assertThat(result.items()).allMatch(f -> f.getTenantId().equals(tenantId));
+  }
+
+  @Test
+  public void shouldValidatePagination() {
+    final var result = zeebeClient.newFlownodeInstanceQuery().page(p -> p.limit(2)).send().join();
+    assertThat(result.items().size()).isEqualTo(2);
+    final var key = result.items().getFirst().getFlowNodeInstanceKey();
+    // apply searchAfter
+    final var resultAfter =
+        zeebeClient
+            .newFlownodeInstanceQuery()
+            .page(p -> p.searchAfter(Collections.singletonList(key)))
+            .send()
+            .join();
+
+    assertThat(resultAfter.items().size()).isEqualTo(6);
+    final var keyAfter = resultAfter.items().getFirst().getFlowNodeInstanceKey();
+    // apply searchBefore
+    final var resultBefore =
+        zeebeClient
+            .newFlownodeInstanceQuery()
+            .page(p -> p.searchBefore(Collections.singletonList(keyAfter)))
+            .send()
+            .join();
+    assertThat(result.items().size()).isEqualTo(2);
+    assertThat(resultBefore.items().getFirst().getFlowNodeInstanceKey()).isEqualTo(key);
+  }
+
+  private static void waitForProcessInstancesToStart(final int expectedProcessInstances) {
+    Awaitility.await("should start process instances and import in Operate")
+        .atMost(Duration.ofSeconds(60))
+        .ignoreExceptions() // Ignore exceptions and continue retrying
+        .untilAsserted(
+            () -> {
+              final var result = zeebeClient.newProcessInstanceQuery().send().join();
+              assertThat(result.items().size()).isEqualTo(expectedProcessInstances);
+            });
+  }
+
+  private static void waitForProcessInstancesToExecute() throws InterruptedException {
+    Thread.sleep(15000);
```

Comment:
❓  Is it possible use Awaitility to check test conditions? 
`Awaitility.await().atMost(Duration.ofSeconds(15)).until(() -> checkCondition());` as it used in other integration tests?

Reply:
Yes, thanks. Copy-paste error.